components/filter/filter.js

```diff
@@ -306,8 +353,6 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 					${this.localize('components.filter.searchResults', { number: count })}
 				</p>
 			`;
-
-			if (count === 0) return searchResults;
```

Comment:
This was a perf enhancement to not render a list where everything is hidden.  But we want to do that now so that we can clear all values.  This happens when a user has selected values in a list and then searches away all values.  When they click the clear button, we still want to clear all those hidden items.